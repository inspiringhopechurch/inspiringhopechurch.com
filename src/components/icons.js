import { library } from "@fortawesome/fontawesome-svg-core";
import {
  faAngleDown,
  faAngleUp,
  faBadgePercent,
  faCommentAltLines,
  faEnvelope,
  faHeart,
  faMapMarkerAlt,
  faUser,
  faPaperPlane,
  faPhone,
  faHeadSideMask,
  faPeopleArrows,
  faPumpSoap,
  faSprayCan,
} from "@fortawesome/pro-solid-svg-icons";
import { faFacebook, faFacebookF, faTwitter, faLinkedinIn, faStripe } from "@fortawesome/free-brands-svg-icons";

// Initialize icon library to use throughout our app
library.add(
  faHeadSideMask,
  faPeopleArrows,
  faPumpSoap,
  faSprayCan,
  faAngleDown,
  faAngleUp,
  faBadgePercent,
  faCommentAltLines,
  faEnvelope,
  faHeart,
  faMapMarkerAlt,
  faUser,
  faPaperPlane,
  faPhone,
  faFacebook,
  faFacebookF,
  faTwitter,
  faLinkedinIn,
  faStripe
);
