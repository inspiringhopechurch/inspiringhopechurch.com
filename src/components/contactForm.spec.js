import React from "react";
import { shallow, mount } from "enzyme";
import ContactForm from "./contactForm";
// Import fontawesome library files
import "./icons";

describe("ContactForm", () => {
  let myForm = undefined;
  const name = "fullName",
    email = "email",
    subject = "messageSubject",
    message = "messageBody";

  const formSubmitButton = "#form-submit",
    testName = "Test User ",
    testEmail = "test@test.com ",
    testSubject = " Test Subject ",
    testMsg = " Test message",
    nameValidationTxt = "Please enter your full name",
    emailValidationTxt = "Please enter your email",
    subjectValidationTxt = "Tell us why you're reaching out in the subject.",
    bodyValidationTxt = "Tell us how we can help you here.";

  beforeEach(() => {
    // myForm = shallow(<ContactForm />);
  });

  it("should render without crashing", () => {
    myForm = shallow(<ContactForm />);
    expect(myForm).toMatchSnapshot();
  });

  it("Form to have default validation messages", () => {
    myForm = mount(<ContactForm />);
    expect(myForm.prop("nameValidationMsg")).toBe(nameValidationTxt);
    expect(myForm.prop("emailValidationMsg")).toBe(emailValidationTxt);
    expect(myForm.prop("subjectValidationMsg")).toBe(subjectValidationTxt);
    expect(myForm.prop("bodyValidationMsg")).toBe(bodyValidationTxt);
  });

  it("disables submit button when required inputs are not filled out", () => {
    myForm = mount(<ContactForm />);
    expect(myForm.find(formSubmitButton).prop("disabled")).toBeTruthy();
  });

  it("enables submit button only when all inputs are filled in", () => {
    myForm = mount(<ContactForm />);

    expect(myForm.find(formSubmitButton).prop("disabled")).toBeTruthy();

    myForm.find(`#${name}`).simulate("change", { target: { name: name, value: testName } });
    myForm.find(`#${email}`).simulate("change", { target: { name: email, value: testEmail } });
    myForm.find(`#${subject}`).simulate("change", { target: { name: subject, value: testSubject } });
    expect(myForm.find(formSubmitButton).prop("disabled")).toBeTruthy();

    myForm.find(`#${message}`).simulate("change", { target: { name: message, value: testMsg } });
    expect(myForm.find(formSubmitButton).prop("disabled")).toBeFalsy();

    // These tests were added when I was trimming spaces from the typed in values.
    // Now, I'm not doing that anymore. Considering if I should be modifying user input,
    // Or instead, give a warning message when there are trailing spaces. For now, disable
    // myForm.find(`#${name}`).simulate("change", { target: { name: name, value: " " } });
    // expect(myForm.find(formSubmitButton).prop("disabled")).toBeTruthy();

    // myForm.find(`#${name}`).simulate("change", { target: { name: name, value: testName } });
    // expect(myForm.find(formSubmitButton).prop("disabled")).toBeFalsy();
  });

  it("changes inputs when values are filled in", () => {
    myForm = mount(<ContactForm />);

    //! We keep querying myForm.find(`selector`) and don't store this in a
    //! variable, because enzyme will not pick up changes if we do this.
    expect(myForm.find(`#${name}`).prop("value").length).toBe(0);
    myForm.find(`#${name}`).simulate("change", { target: { name: name, value: testName } });
    expect(myForm.find(`#${name}`).prop("value")).toEqual(testName);
    expect(myForm.find(`#${name}`).prop("value").length).toBe(testName.length);

    expect(myForm.find(`#${email}`).prop("value").length).toBe(0);
    myForm.find(`#${email}`).simulate("change", { target: { name: "email", value: testEmail } });
    expect(myForm.find(`#${email}`).prop("value")).toEqual(testEmail);
    expect(myForm.find(`#${email}`).prop("value").length).toBe(testEmail.length);

    expect(myForm.find(`#${subject}`).prop("value").length).toBe(0);
    myForm.find(`#${subject}`).simulate("change", {
      target: { name: "messageSubject", value: testSubject },
    });
    expect(myForm.find(`#${subject}`).prop("value")).toEqual(testSubject);
    expect(myForm.find(`#${subject}`).prop("value").length).toBe(testSubject.length);

    expect(myForm.find(`#${message}`).text().length).toBe(0);
    myForm.find(`#${message}`).simulate("change", {
      target: { name: message, value: testMsg },
    });
    myForm.find(`#${message}`).simulate("change");
    // Getting the value of *this* input a different way, just because I can
    expect(myForm.find(`#${message}`).getDOMNode().value.length).toBe(testMsg.length);
  });
});
