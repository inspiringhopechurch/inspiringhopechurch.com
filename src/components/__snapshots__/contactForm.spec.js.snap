// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactForm should render without crashing 1`] = `
<div
  className="contact-form content"
>
  <FancyHeading
    className="has-text-centered"
    heading="Get in touch"
  />
  <div
    className="notification "
  >
    <button
      aria-label="remove notification"
      className="delete"
      onClick={[Function]}
    />
  </div>
  <form
    data-testid="contact-form"
    id="contact-form"
    method="post"
    onSubmit={[Function]}
  >
    <div
      className="field"
    >
      <div
        className=""
      >
        <div
          className="field"
        >
          <label
            className="label"
            htmlFor="fullName"
          >
            Name
          </label>
          <div
            className="control is-expanded has-icons-left"
          >
            <input
              className="input"
              data-testid="fullName"
              id="fullName"
              name="fullName"
              onChange={[Function]}
              placeholder="Full Name"
              required={true}
              type="text"
              value=""
            />
            <span
              className="icon is-left is-small"
            >
              <FontAwesomeIcon
                beat={false}
                beatFade={false}
                border={false}
                bounce={false}
                className=""
                fade={false}
                fixedWidth={false}
                flip={null}
                icon={
                  Array [
                    "fas",
                    "user",
                  ]
                }
                inverse={false}
                listItem={false}
                mask={null}
                maskId={null}
                pull={null}
                pulse={false}
                rotation={null}
                shake={false}
                size="1x"
                spin={false}
                swapOpacity={false}
                symbol={false}
                title=""
                titleId={null}
                transform={null}
              />
            </span>
          </div>
          <p
            className="help"
            id="fullNameError"
          >
             
          </p>
        </div>
        <div
          className="field"
        >
          <label
            className="label"
            htmlFor="email"
          >
            Email
          </label>
          <div
            className="control is-expanded has-icons-left"
          >
            <input
              className="input"
              data-testid="email"
              id="email"
              name="email"
              onChange={[Function]}
              placeholder="Email Address"
              required={true}
              type="email"
              value=""
            />
            <span
              className="icon is-left is-small"
            >
              <FontAwesomeIcon
                beat={false}
                beatFade={false}
                border={false}
                bounce={false}
                className=""
                fade={false}
                fixedWidth={false}
                flip={null}
                icon={
                  Array [
                    "fas",
                    "envelope",
                  ]
                }
                inverse={false}
                listItem={false}
                mask={null}
                maskId={null}
                pull={null}
                pulse={false}
                rotation={null}
                shake={false}
                size="1x"
                spin={false}
                swapOpacity={false}
                symbol={false}
                title=""
                titleId={null}
                transform={null}
              />
            </span>
          </div>
          <p
            className="help"
            id="emailError"
          >
             
          </p>
        </div>
      </div>
    </div>
    <div
      className="field"
    >
      <label
        className="label"
        htmlFor="messageSubject"
      >
        Subject
      </label>
      <div
        className="control has-icons-left"
      >
        <input
          className="input"
          data-testid="messageSubject"
          id="messageSubject"
          name="messageSubject"
          onChange={[Function]}
          placeholder="Email Subject Line"
          required={true}
          type="text"
          value=""
        />
        <span
          className="icon is-left is-small"
        >
          <FontAwesomeIcon
            beat={false}
            beatFade={false}
            border={false}
            bounce={false}
            className=""
            fade={false}
            fixedWidth={false}
            flip={null}
            icon={
              Array [
                "fas",
                "comment-alt",
              ]
            }
            inverse={false}
            listItem={false}
            mask={null}
            maskId={null}
            pull={null}
            pulse={false}
            rotation={null}
            shake={false}
            size="1x"
            spin={false}
            swapOpacity={false}
            symbol={false}
            title=""
            titleId={null}
            transform={null}
          />
        </span>
      </div>
      <p
        className="help"
        id="messageSubjectError"
      >
         
      </p>
    </div>
    <div
      className="field"
    >
      <label
        className="label"
        htmlFor="messageBody"
      >
        Message
      </label>
      <div
        className="control"
      >
        <textarea
          className="textarea"
          data-testid="messageBody"
          id="messageBody"
          name="messageBody"
          onChange={[Function]}
          placeholder="Send us your message here."
          required={true}
          rows={10}
          value=""
        />
      </div>
      <p
        className="help"
        id="messageBodyError"
      >
         
      </p>
    </div>
    <div
      className="field"
    >
      <div
        className="control"
      >
        <button
          className="button is-link is-outlined is-fullwidth"
          data-testid="form-submit"
          disabled={true}
          id="form-submit"
          type="submit"
        >
          <span
            className="icon"
          >
            <FontAwesomeIcon
              beat={false}
              beatFade={false}
              border={false}
              bounce={false}
              className=""
              fade={false}
              fixedWidth={false}
              flip={null}
              icon={
                Array [
                  "fas",
                  "paper-plane",
                ]
              }
              inverse={false}
              listItem={false}
              mask={null}
              maskId={null}
              pull="left"
              pulse={false}
              rotation={null}
              shake={false}
              size="1x"
              spin={false}
              swapOpacity={false}
              symbol={false}
              title=""
              titleId={null}
              transform={null}
            />
          </span>
          <span>
            Send Message
          </span>
        </button>
      </div>
    </div>
  </form>
</div>
`;
